scripts/pre_push_hook.py

```diff
@@ -133,48 +157,45 @@ def _collect_files_being_pushed(ref_list, remote):
     branches = [ref.local_ref.split('/')[-1] for ref in ref_heads_only]
     hashes = [ref.local_sha1 for ref in ref_heads_only]
     remote_hashes = [ref.remote_sha1 for ref in ref_heads_only]
-    modified_files = set()
-    files_to_lint = set()
+    collected_files = {}
     for branch, sha1, remote_sha1 in zip(branches, hashes, remote_hashes):
         # git reports the following for an empty / non existing branch
         # sha1: '0000000000000000000000000000000000000000'
         if set(remote_sha1) != {'0'}:
             try:
-                file_diffs = _compare_to_remote(remote, branch)
+                modified_files = _compare_to_remote(remote, branch)
             except ValueError as e:
                 print e.message
                 sys.exit(1)
             else:
-                modified_files.update(file_diffs)
-                files_to_lint.update(_extract_files_to_lint(modified_files))
+                files_to_lint = _extract_files_to_lint(modified_files)
```

Comment:
The issue is that I'm using it as a flow control. 

I try to compare the branch being pushed with the same branch on the remote,  if this fails go to fallback (check against develop) 

Issue is I only know if I failed after I tried